.github/workflows/scripts/e2e.gradle.default.verify.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# shellcheck source=/dev/null
+source "./.github/workflows/scripts/e2e-verify.common.sh"
+
+# Input variables
+EXPECTED_ARTIFACT_OUTPUT=${EXPECTED_ARTIFACT_OUTPUT:-}
+PROVENANCE_DIR=${PROVENANCE_DIR:-}
+GITHUB_REF_NAME=${GITHUB_REF_NAME:-}
+GITHUB_REF=${GITHUB_REF:-}
+GITHUB_REF_TYPE=${GITHUB_REF_TYPE:-}
+RUNNER_DEBUG=${RUNNER_DEBUG:-}
+if [[ -n "${RUNNER_DEBUG}" ]]; then
+    set -x
+fi
+
+cp -r  "slsa-attestations" "${PROJECT_DIR}/"
+cp -r build "${PROJECT_DIR}/"
+cd "${PROJECT_DIR}"
+
+artifact_version=$(./gradlew properties -q | grep "version:" | awk '{print $2}')
+artifact_id=$(./gradlew properties -q | grep "name:" | awk '{print $2}')
+artifact_name="${artifact_id}-${artifact_version}.jar"
+provenance="${PROVENANCE_DIR}/${artifact_name}.build.slsa"
+
+go env -w GOFLAGS=-mod=mod
+
+verify_provenance_content() {
+    local attestation
+    attestation=$(jq -r '.dsseEnvelope.payload' "${provenance}" | base64 -d)
+
+    # Run the artifact and verify the output is correct
+    artifact_output=$(java -jar build/libs/"${artifact_name}")
+    expected_artifact_output="${EXPECTED_ARTIFACT_OUTPUT}"
+    e2e_assert_eq "${artifact_output}" "${expected_artifact_output}" "The output from the artifact should be '${expected_artifact_output}' but was '${artifact_output}'"
+    
+    # Verify the content of the attestation
+    e2e_verify_predicate_subject_name "${attestation}" "${artifact_name}"
+    e2e_verify_predicate_v1_runDetails_builder_id "${attestation}" "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/builder_gradle_slsa3.yml@refs/heads/main"
+    e2e_verify_predicate_v1_buildDefinition_buildType "${attestation}" "https://github.com/slsa-framework/slsa-github-generator/delegator-generic@v0"
```

Comment:
nit: run the binary and verify the output message, like for Maven builder

Reply:
It already does that above on these lines:

```bash
    # Run the artifact and verify the output is correct
    artifact_output=$(java -jar build/libs/"${artifact_name}")
    expected_artifact_output="${EXPECTED_ARTIFACT_OUTPUT}"
    e2e_assert_eq "${artifact_output}" "${expected_artifact_output}" "The output from the artifact should be '${expected_artifact_output}' but was '${artifact_output}'"
```